internal/pkg/describe/ecs_service_status.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package describe
+
+import (
+	"strings"
+
+	"github.com/aws/copilot-cli/internal/pkg/aws/ecs"
+	"github.com/aws/copilot-cli/internal/pkg/aws/elbv2"
+)
+
+func allContainerHealthEmpty(tasks []ecs.TaskStatus) bool {
+	// A container health is considered empty if it is empty or UNKNOWN
+	for _, t := range tasks {
+		if t.Health != ecs.TaskContainerHealthStatusUnknown && t.Health != "" {
+			return false
+		}
+	}
+	return true
+}
+
+func allCapacityProviderEmpty(tasks []ecs.TaskStatus) bool {
+	for _, task := range tasks {
+		if task.CapacityProvider != "" {
+			return false
+		}
+	}
+	return true
+}
+
+func anyTaskATarget(tasks []ecs.TaskStatus, targetsHealth []taskTargetHealth) bool {
+	taskToHealth := summarizeHTTPHealthForTasks(targetsHealth)
+	for _, t := range tasks {
+		if _, ok := taskToHealth[t.ID]; ok {
+			return true
+		}
+	}
+	return false
+}
+
+func containerHealthDataForTasks(tasks []ecs.TaskStatus) (healthy int, unhealthy int, unknown int) {
+	for _, t := range tasks {
+		switch strings.ToUpper(t.Health) {
+		case ecs.TaskContainerHealthStatusHealthy:
+			healthy += 1
+		case ecs.TaskContainerHealthStatusUnhealthy:
+			unhealthy += 1
+		case ecs.TaskContainerHealthStatusUnknown:
+			unknown += 1
+		}
+	}
+	return
+}
+
+func healthyHTTPTaskCountInTasks(tasks []ecs.TaskStatus, targetsHealth []taskTargetHealth) int {
```

Comment:
nit:
```suggestion
func countHealthyHTTPTasks(tasks []ecs.TaskStatus, targetHealthDescriptions []taskTargetHealth) int {
```
I find reading `targetsHealth` a bit strange 😛 what do you think of using `Descriptions` it's a bit more verbose but less awkward I think?